File: play-2.4.x/K-Modules/kmodules/K-Base/app/K/Aop/annotations/CheckToken.java
package K.Aop.annotations;

import play.mvc.With;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kk on 13-12-16.
 */

@With(K.Aop.actions.CheckTokenAction.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface CheckToken {
    String token_arg() default "token";

    int[] allowed_roles() default {};
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Aop/annotations/CheckToken.kt
package K.Aop.annotations

import play.mvc.With
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by kk on 13-12-16.
 */

@With(K.Aop.actions.CheckTokenAction::class)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class CheckToken(val token_arg: String = "token", val allowed_roles: IntArray = intArrayOf())
